rust/src/encodings/plain.rs

```diff
@@ -129,13 +130,16 @@ impl<'a> PlainDecoder<'a> {
             DataType::Boolean => bit_util::ceil(self.length, 8),
             _ => get_primitive_byte_width(self.data_type)? * self.length,
         };
-        let range = Range {
-            start: self.position,
-            end: self.position + array_bytes,
-        };
+        let mut start = self.position;
```

Comment:
should we make the other encoders do this as well?

Reply:
Yes, certainly.